### ISSUE TITLE
Integration with Moloch

### ISSUE THREAD

**Issue by botherder** [cuckoobox/cuckoo#523]
Moloch is a powerful network capture and indexing system which is increasingly commonly used in corporate networks to monitor traffic and identify malicious activity.
By integrating Cuckoo and Moloch, we'd be able to create a fully open source sandnet capable of extracting malicious artifacts on-the-fly and requesting Cuckoo to analyze them.

We won't be able to achieve this without coordinating heavily with the Moloch team, so we should get @WilliamSalusky into the discussion.

**Comment by eoinmiller** [comment]
I know @wmetcalf uses Cuckoo + Moloch quite a bit. Maybe he can share some info on how EmergingThreat uses them together currently.

**Comment by rep** [comment]
Disagree unless I misunderstand the goal/motivation for using Moloch.

First of all "integration" would merely mean running moloch-capture on the interface that is used for the VM traffic - there's not much to do on top of that unless you want to add some sort of tagging that labels the streams in Moloch by which VM or which malware sample was running at the time.

Also Moloch overall is mostly a search/browsing tool for the traffic, it doesn't really offer any analysis or scripting features for extracting certain details (as far as I know, please correct me). Thus - no offense - it would be mostly a nicer version of our current network analysis interface. Having to run Moloch + Elasticsearch just for the nicer browsing feature of the traffic seems quite a bit heavy to me.

Again I'm a bit unclear about the design goal. I used Moloch for some time but in the end decided it was too heavyweight for my purposes as well. Apologies to Moloch, I think it's a great tool, just not so much for automation purposes :)

**Comment by botherder** [comment]
You're missing out the whole point, Mark.

The point isn't to integrate Moloch INTO Cuckoo, it is to integrate the two tools to work in parallel. Moloch is used a full take network monitoring system on corporate networks and it is able to extract interesting information or artifacts.

With Moloch sitting in front of a corporate network, it would be capable of for example pulling executables downloaded and submitting them automatically to Cuckoo to perform an analysis. Perhaps it would then be able to pull some brief results from the Cuckoo instance and display them in the Moloch interface providing links to the analyses.

This has nothing to do with analysis of network traffic from Cuckoo's analyses.

**Comment by bit-bucket** [comment]
I am interested in this.

**Comment by KillerInstinct** [comment]
@botherder @rep I think this would be sort-of. As Moloch is an indexing service, it may be more valuable to implement the ability to use ElasticSearch in place of Mongo (so the user can choose to do either or) Personally, for my research and dev purposes I have no need for Moloch with my cuckoo instance. Also, pretty much all of the functionality can be replicated with other (prettier) tools such as Kibana/Splunk/Etc. IMO I think it would be easier and more beneficial to make a reporting module that reports out data the Moloch would want, and supply it to another system -- EG Splunk/Elastic+Kibana, etc. Moloch overall seems quite difficult to truly integrate with, outside of running Cuckoo + Moloch and creating hyperlinks in Django. If Cuckoo ever got an auth system, it wouldn't be compatible to Moloch without upstream changes. Both projects are really quite different...

**Comment by botherder** [comment]
For now we close the ticket. It seems premature from both sides.

**Comment by jbremer** [comment]
Why not just postpone it to 2.1 or similar?
